dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -423,9 +428,14 @@ def silent_takeover(oauth_user, auth_hash)
       )
       unless success
```

Comment:
user.update appears to behave as you describe it when I try it in rails console, returning false here if validations fail. however, the [docs](https://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-update) say that `update` returns the object, rather than a boolean. am I looking at the wrong docs?

Reply:
same question here I guess, why not `update!`?

Reply:
It looks like the documented return value of `update` on failure has changed between [our Rails version (5.0.1)](https://api.rubyonrails.org/v5.0.1/classes/ActiveRecord/Persistence.html#method-i-update) and the [latest (5.2.1)](https://api.rubyonrails.org/v5.2.1/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-update)!  This surprises me, because that seems like it breaks semver.

Switching to `update!` and using a rescue block to customize our Honeybadger notification makes sense - I'll do that.

Reply:
Thanks for the suggestion Dave, I like how this looks with `create!`/`update!`/`rescue` much better, even though at first glance it's a messier diff.  PTAL!